File: Competitive Programming 3/Section 1.4/Games (Others), Easier/10279 - Mine Sweeper/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in);
    static int[][] movs = {{1,0},{0,1},{-1,0},{0,-1},{1,1},{-1,-1},{-1,1},{1,-1}};
    static char getAns(char c,boolean showMines){
        if(c=='*')
            return showMines?'*':'.';
        return c;
    }
    static int convert(char c){
        return c=='*'?1:0;
    }
    public static void main(String[]args){
        int t = Integer.parseInt(s.nextLine()),n;
        for(int cs=0;cs<t;cs++){
            if(cs>0)
                System.out.printf("\n");
            s.nextLine();
            n=Integer.parseInt(s.nextLine());
            char[][]grid=new char[n+2][n+2];
            for(int i=1;i<=n;i++){
                String line = s.nextLine();
                for(int j=1;j<=n;j++)
                    grid[i][j]=line.charAt(j-1);
            }
            boolean showMines=false;
            for(int i=1;i<=n;i++){
                String line = s.nextLine();
                for(int j=1;j<=n;j++){
                    if(line.charAt(j-1)=='x')//touched
                        if(grid[i][j]=='*') //bomb
                            showMines=true;
                        else {//not bomb, sum up adjacents
                            grid[i][j]='0';
                            for(int mov=0;mov<8;mov++)
                                grid[i][j]+=convert(grid[i+movs[mov][0]][j+movs[mov][1]]);
                        }
                }
            }
            for(int i=1;i<=n;i++){
                for(int j=1;j<=n;j++){
                    System.out.printf("%c",getAns(grid[i][j],showMines));
                }
                System.out.printf("\n");
            }
        }
    }
}

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/10279 - Mine Sweeper/Python/main.py
movs=[[1,0],[0,1],[-1,0],[0,-1],[1,1],[-1,-1],[-1,1],[1,-1]]
def getAns(c,showMines):
    if c=='*':
        return '*' if showMines else '.'
    return c
def convert(c):
    return 1 if c=='*' else 0 
def main():
    for cs in range(int(input())):
        input()
        n = int(input())
        if cs > 0:
            print()
        grid=[['.' for _ in range(n+2)] for _ in range(n+2)]
        for i in range(1,n+1):
            for j,el in enumerate(input(),1):
                grid[i][j]=el
        showMines=False
        for i in range(1,n+1):
            for j,el in enumerate(input(),1):
                if el=='x': #touched
                    if grid[i][j]=='*': #bomb
                        showMines=True
                    else: #not bomb , sum up adjacents
                        grid[i][j]=ord('0')
                        for mov in movs:
                            grid[i][j]+=convert(grid[i+mov[0]][j+mov[1]])
                        grid[i][j]=chr(grid[i][j])
        for i in range(1,n+1):
            for j in range(1,n+1):
                print(getAns(grid[i][j],showMines),end='')
            print()
main()